scripts/schema.bash

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+# Copyright 2019 Canonical Ltd.
+# Licensed under the AGPLv3, see LICENCE file for details.
+function schema_sha {
+    echo `cat ./apiserver/facades/schema.json | sha256sum | awk '{ print $1 }'`
+}
+
+current_schema_sha=$(schema_sha)
+make rebuild-schema
+new_schema_sha=$(schema_sha)
+if [ $current_schema_sha != $new_schema_sha ]; then
+    (>&2 echo "Error: facades schema is not in sync. Run `make rebuild-schema` and commit source.")
+fi
```

Comment:
missing newline at end of file.
Also, I'm not super happy with checking the content on disk before you check the content on disk. (consider verify.bash being used pre-push). Doesn't it need to verify against the value of the last commit?
I think you could use "git show HEAD -- apiserver/facades/schema.json" or something like that to get the last committed version. Thoughts?
I would have thought there was a way to just get the sha from git but I don't see it (AIUI git commit hashes have more than just content.)

Reply:
The check in how I envisioned it was, "it looks like you're attempting push the schema.json which is out of line with what the go code states". The problem is that if we use the last commit of git to test against, what are we trying to determine; if you changed a file, what was the reason, did you change the API server facades? The only true way to really know if something changed is to ask the facades themselves...

We could do something along the lines of the golinters and ignore them for local pre-push (unless you set the argument) and enable them for CI, that way you don't have to take the hit of rebuild the schema for every push.

Reply:
> Doesn't it need to verify against the value of the last commit?

So the schema.json will be the last commit if it's done on in pre-push, but I can see that we could be sure and use `git show HEAD:./apiserver/facades/schema.json`, as we then know that what you're pushing is in fact what we need to check.

Reply:
I've address the issues in https://github.com/juju/juju/pull/10305/commits/0328e7276a101144a77b9b0c92bbac94109b211a